src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/matcher/RegionMatcher.java

```diff
@@ -33,9 +34,11 @@ public boolean matchAttribute(RoleAssignment roleAssignment, CaseDetails caseDet
     }
 
     private Optional<String> getRegion(CaseDetails caseDetails) {
-        JsonNode caseManagementLocation = caseDetails.getData().get(CASE_MANAGEMENT__LOCATION);
-        if (caseManagementLocation != null && caseManagementLocation.get(REGION) != null) {
-            return Optional.ofNullable(caseManagementLocation.get(REGION).asText());
+        if (caseDetails != null && caseDetails.getData() != null) {
```

Comment:
This will avoid NPE, but It looks to me like...
If caseDetails is null, there should be an error handling for it in the first instance.
The method ideally shouldn't be garbage argument (null) and therefore it ideally shouldn't need NP check.